modules/ROOT/pages/dataweave-type-selector.adoc

```diff
@@ -0,0 +1,206 @@
+ userName : String
+}
+type Address = User.address
+
+var userAddress: Address = {country: "Argentina", city: "Rosario", street: "Calle Falsa", number: 123}
+---
+userAddress
+----
+
+.Output:
+[source,json,linenums]
+----
+{
+ "country": "Argentina",
+ "city": "Rosario",
+ "street": "Calle Falsa",
+ "number": 123
+}
+----
+
+=== Union Types
+The next example shows how type selection can also be used with `Union` types that consist of `Object` types:
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+type ID = {name: String} | {id: Number}
+var name: ID = {name: "Seba"}
+var id: ID = {id: 38123}
+var nameString: ID.name = "Seba"
+var idNumber: ID.id = 38123
+---
+[name, id, nameString ++ " is of type " ++ typeOf(nameString), idNumber ++ " is of type " ++ typeOf(idNumber)]
+----
+
+.Output:
+[source,json,linenums]
+----
+[
+  {
+    "name": "Seba"
+  },
+  {
+    "id": 38123
+  },
+  "Seba is of type String",
+  "38123 is of type Number"
+]
+----
+
+However, the following script fails because the second type is not an `Object` type and there is no way to reference it with the selector:
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+type UnionType = {name: String} | Number
+type Fail = UnionType.name
+---
+{}
+----
+
+.Output:
+[source,json,linenums]
+----
+Cannot do selection on Type: Number
+
+4| type Fail = UnionType.name
+----
+
+=== Namespace
+The following example shows type selection of a namespace:
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+ns ns1 http://acme.com
+type User = {
+ ns1#name : String,
+ name : Number,
+}
+var nameString: User.ns1#name = "Seba"
+var nameNumber: User.name = 123
+---
+{
+   "NameString" : typeOf(nameString),
+   "NameNumber" : typeOf(nameNumber)
+}
+----
+
+.Output:
+[source,json,linenums]
+----
+{
+ "NameString": "String",
+ "NameNumber": "Number"
+}
+----
+
+=== Metadata
+This example outlines how type selection preserves metadata associated with the type:
+
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+type User = {
+ birthDate: Date {format: "dd-MMM-yy"},
+ userName : String {schema: "value"}
+}
+type FormattedDate = User.birthDate
+type UserName = User.userName
+var formattedDate: FormattedDate = "10-SEP-15" as Date {format: "dd-MMM-yy"}
+var otherFormatDate = "23-10-2022" as Date {format: "dd-MM-yyyy"}
+var userName = "Messi" as String {schema: "value"}
+var otherUserName = "Di María" as String {schema: "otherValue"}
+---
+{
+formattedDate: formattedDate is FormattedDate,
+userName: userName is UserName,
+otherFormatDate: otherFormatDate is FormattedDate,
+otherUserName: otherUserName is UserName
+}
+----
+
+.Output:
+[source,json,linenums]
+----
+{
+ "formattedDate": true,
+ "userName": true,
+ "otherFormatDate": false,
+ "otherUserName": false
+}
+----
+
+=== Generics
```

Comment:
we're calling these "type parameters" 
See "Specify Type Parameters" here: https://github.com/mulesoft/docs-dataweave/blob/19460483e6162827f56e86e95ea284ba4825a182/modules/ROOT/pages/dataweave-functions-lambdas.adoc

Reply:
Same thing to modify the section using the H2 pattern. 

== Generics

== Selecting Types with Type Parameters (Generics)

or something like that maybe?